lib/modules/datasource/index.ts

```diff
@@ -172,24 +175,64 @@ async function mergeRegistries(
       }
 
       if (!combinedRes) {
+        // This is the first registry, so we can just use it and continue
         combinedRes = res;
         continue;
       }
 
-      if (commonRegistryUrl) {
+      if (singleRegistry) {
+        // This is the second registry
+        // We need to move the registryUrl from the package level to the release level
         for (const release of coerceArray(combinedRes.releases)) {
           release.registryUrl ??= combinedRes.registryUrl;
         }
-        commonRegistryUrl = false;
+        singleRegistry = false;
       }
 
       const releases = coerceArray(res.releases);
       for (const release of releases) {
+        // We have more than one registry, so we need to move the registryUrl
+        // from the package level to the release level before merging
         release.registryUrl ??= res.registryUrl;
       }
 
       combinedRes.releases.push(...releases);
+
+      // Merge the tags from the two results
+      let tags = combinedRes.tags;
+      if (tags) {
+        if (res.tags) {
+          // Both results had tags, so we need to compare them
+          for (const tag of ['release', 'latest']) {
+            const existingTag = combinedRes?.tags?.[tag];
+            const newTag = res.tags?.[tag];
+            if (is.string(newTag) && mavenVersioning.isVersion(newTag)) {
```

Comment:
so this combining is only done for maven? It would also happen if another datasource is used and the versions are maven compatible.

Should we guard this to only work on maven datasource?

nuget has also hunt strategy, but doesn't return tags yet. it may in future and nobody would probably think about this maven thing here.

Reply:
Good catch, I changed it to be generic